src/applySpec.js

```diff
@@ -0,0 +1,40 @@
+var _curry1 = require('./internal/_curry1');
+var curryN = require('./curryN');
+var map = require('./map');
+var max = require('./max');
+var pluck = require('./pluck');
+var reduce = require('./reduce');
+var values = require('./values');
+
+
+/**
+ * Given a spec object recursively mapping properties to functions, creates a
+ * function producing an object of the same structure, by mapping each property
+ * to the result of calling its associated function with the supplied arguments.
+ *
+ * @func
+ * @memberOf R
+ * @category Function
+ * @sig {k: ((a, b, ..., m) -> v)} -> ((a, b, ..., m) -> {k: v})
+ * @param {Object} spec an object recursively mapping properties to functions for
+ *        producing the values for these properties.
+ * @return {Function} A function that returns an object of the same structure
+ * as `spec', with each property set to the value returned by calling its
+ * associated function with the supplied arguments.
+ * @example
+ *
+ *      var getMetrics = R.applySpec({
+ *                                      sum: R.add,
+ *                                      nested: { mul: R.multiply }
+ *                                   });
+ *      getMetrics(2, 4); // => { sum: 6, nested: { mul: 8 } }
+ */
+module.exports = _curry1(function applySpec(spec) {
+  spec = map(function(v) { return typeof v == 'function' ? v : applySpec(v) },
+             spec);
+  return curryN(reduce(max, 0, pluck('length', values(spec))),
+                function() {
+                  var args = arguments;
+                  return map(function(f) { return f.apply(null, args); }, spec);
```

Comment:
Sure, will do in the round.

By the way, I must say that this line is confusing:
https://github.com/ramda/ramda/blob/master/src/apply.js#L24

If it's intentionally trying to support mapping ``this`` to the global object, I guess that's fine*, but it better be documented (in the code), because it's easy to overlook that. 

_______
* Note that this makes the caller environment significant:

```js
(function f() { return require("ramda").apply(function() { return this; }, []);  })(); // => Node's global object;
(function f() { "use strict"; return require("ramda").apply(function() { return this; }, []);  })() // => undefined
```